### PR TITLE
[MM-44560] Review plugin's configuration settings

### PR DESCRIPTION
**Pull request by streamer45**
#### Summary

PR reviews Calls plugin's configuration settings:

- Reworked the order and labeling of settings. 
- Prepending `(Optional)` to help text of non mandatory settings.
- Added a header with some description and link to documentation. Please let me know if we should also link on a per-setting basis. 
- Changed the plugin's display name to include the Beta word.
- Calls are now disabled by default unless MM is running on a Cloud license.

Please note that I've purposely deferred adding a setting to configure max participants per call to a follow-up ticket ([MM-44651](https://mattermost.atlassian.net/browse/MM-44651)) as it's a slightly more involved piece of work.

#### Screenshots

image

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-44560

### CODE REVIEW

File: plugin.json [hunk 1]
```diff
@@ -36,33 +30,40 @@
           "placeholder": "8443"
         },
         {
-          "key": "RTCDServiceURL",
-          "display_name": "RTCD Service URL",
+          "key": "AllowEnableCalls",
+          "display_name": "Enable on Specific Channels",
+          "type": "bool",
+          "help_text": "When set to true, it allows channel admins to enable or disable calls in their channels. It also allows participants of DMs/GMs to enable or disable calls.",
```
Comment (justinegeffen) [inline plugin.json:36]:
```suggestion
          "help_text": "When set to true, Channel Admins can enable or disable calls in their channels. It also allows participants in DMs/GMs to enable or disable calls.",
```

File: plugin.json [hunk 2]
```diff
@@ -36,33 +30,40 @@
...
+          "default": false
+        },
+        {
+          "key": "DefaultEnabled",
+          "display_name": "Enable on All Channels",
```
Comment (justinegeffen) [inline plugin.json:41]:
```suggestion
          "display_name": "Enable on all channels",
```

File: plugin.json [hunk 3]
```diff
@@ -36,33 +30,40 @@
...
+          "type": "bool",
+          "help_text": "When set to true, calls will be possible in all channels where they are not explicitly disabled.",
```
Comment (justinegeffen) [inline plugin.json:43]:
```suggestion
          "help_text": "When set to true, calls can be started in all channels where they're not explicitly disabled.",
```

File: plugin.json [hunk 4]
```diff
@@ -36,33 +30,40 @@
...
-          "placeholder": "https://rtcd.example.com"
+          "help_text": "(Optional) The IP (or hostname) to be used as the host ICE candidate. If empty, it defaults to resolving via STUN.",
+          "default": ""
         },
         {
           "key": "ICEServers",
           "display_name": "ICE Servers",
           "type": "text",
-          "help_text": "A comma separated list of ICE servers URLs (STUN/TURN) to use.",
+          "help_text": "(Optional) A comma separated list of ICE servers URLs (STUN/TURN) to use.",
```
[the remark this answers is not in the document]
Comment (justinegeffen) [reply plugin.json:57]: To confirm: Is this a list of URLs belonging to ICE servers? Or a URL belonging to a specific ICE server?

File: plugin.json [hunk 5]
```diff
@@ -36,33 +30,40 @@
...
-          "default": false
-        },
-        {
-          "key": "DefaultEnabled",
-          "display_name": "Default Enabled Calls",
-          "type": "bool",
-          "help_text": "When set to true, calls will be possible in all channels where they are not explicitly disabled.",
-          "default": true
+          "key": "RTCDServiceURL",
+          "display_name": "RTCD Service URL",
```
Comment (justinegeffen) [inline plugin.json:63]:
```suggestion
          "display_name": "RTCD service URL",
```

**Review by justinegeffen — COMMENTED** [review]
I've changed the casing of UI elements to sentence case to match what we've done in the documentation. This change hasn't yet been implemented across the System Console, so this will be inconsistent with other screens for a short while.

Added one question.

File: plugin.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (streamer45) [reply plugin.json:57]: Each ICE Server is identified by a URL and users can specify more than one through this list. Does this make sense?

**Pull request merged**